### PR TITLE
docs(pie-checkbox): DSW-2223 motion tab for checkbox docs

### PR DESCRIPTION
**Pull request by dandel10n**
## Describe your changes (can list changeset entries if preferable)

---
"pie-docs": minor
---

[Added] - a new contentWrapper shortcode to visually combine different items into a visual container. So far it has only one design (white background with 2px rounded border and 24px padding).
[Added] - a new motion tab for pie-checkbox.
[Changed] - moved code section to its own folder under pie-checkbox.

🎨  [Figma Design ](https://www.figma.com/design/OOgnT2oNMdGFytj5AanYvt/%5BCore%5D-Web-Component-Documentation-%5BPIE-3%5D?node-id=5787-11448&t=k8IuIWAT85bkhBIA-4)
🥧  [Feature url](https://pr1723-docs.pie.design/components/checkbox/motion/)

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the Docs site preview
- [x] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1 @leksaBoiko 
- [x] I have reviewed the Docs site preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 @maledr5 
- [X] I have reviewed the Docs site preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Comment by leksaBoiko** [comment]
The timing dots on both desktop and mobile are being partially cut off: 

Screenshot 2024-08-22 at 14 46 37
Screenshot 2024-08-22 at 14 36 42

**Review by leksaBoiko — APPROVED** [review]

**Comment by maledr5** [comment]
Everything looks good. I found that in the Figma design the `Unselected` notes for `Tick` property are different than the implementation, is that an error in the design or we need to update the json file? 
Screenshot 2024-08-23 at 10 52 46

**Comment by dandel10n** [comment]
> Everything looks good. I found that in the Figma design the `Unselected` notes for `Tick` property are different than the implementation, is that an error in the design or we need to update the json file? Screenshot 2024-08-23 at 10 52 46

I updated the starting scale value to follow the doc

**Review by maledr5 — APPROVED** [review]

File: apps/pie-docs/src/components/checkbox/motion/motion.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        alt: "A chart showing timing in milliseconds for hover 01 property."
+    } %}
+    {% componentDetailsTable { tableData: hover } %}
+{% endcontentWrapper %}
+
+---
+
+## Active
+
+When the user clicks a checkbox (on:Click), the active state $active01 fades in at 100ms and then out at 100ms.
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/checkbox/motion/motion.md:43]: can we wrap any code-style copy in `` for better visual clarity? So in this case `$active01`

File: apps/pie-docs/src/components/checkbox/motion/motion.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    alt: "Three checkboxes in a horizontal row. First one in default state, second one in hover state and third one in checked state.",
+    width: "586px",
+    height: "m"
+} %}
+
+---
+
+## Hover
+
+When the user hovers over an unselected checkbox (on:Hover), the hover state $hover01 fades in at 200ms. When a user moves the mouse away (off:Hover) the hover state $hover01 fades out at 200ms.
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/checkbox/motion/motion.md:29]: is `(on:Hover)` our way of indicating browser events? I am not sure if we've agreed on a convention for this before but it's certainly an interesting idea if so. I am unsure what exactly it adds here personally but keen to know what others think.

Comment (xander-marjoram) [reply apps/pie-docs/src/components/checkbox/motion/motion.md:29]: I'm not sure who it's aimed for but to my eye it looks like `:hover` but with incorrect casing 😅

File: apps/pie-docs/src/components/checkbox/motion/motion.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+eleventyComputed:
+    hover: "{% include './hover.json' %}"
+    active: "{% include './active.json' %}"
+    selected: "{% include './selected.json' %}"
+    unselected: "{% include './unselected.json' %}"
+---
+
+## Example
+
+Illustrates the animations that happen in the checkbox component. In this component we have animations on hover and active states as well as when the component is selected and unselected.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/checkbox/motion/motion.md:16]:
```suggestion
Illustrates the animations that happen in the checkbox component. In this component we have animations on hover and active states as well as when the component is selected and deselected.
```

File: apps/pie-docs/src/components/checkbox/motion/motion.md (hunk 2 above)
Comment (jamieomaguire) [reply apps/pie-docs/src/components/checkbox/motion/motion.md:29]: yes, agreed 😂

File: apps/pie-docs/src/components/checkbox/motion/motion.md (hunk 3 above)
Comment (jamieomaguire) [reply apps/pie-docs/src/components/checkbox/motion/motion.md:16]: @xander-marjoram would this also be more appropriate for other usages of 'unselected'?

File: apps/pie-docs/src/components/checkbox/motion/motion.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    {% componentDetailsTable {
+    tableData: selected
+    } %}
+{% endcontentWrapper %}
+
+---
+
+## Unselected
+
+When a checkbox is unselected both the tick and the fill scale down 100-0% at 100ms.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/checkbox/motion/motion.md:75]:
```suggestion
When a checkbox is deselected both the tick and the fill scale down 100-0% at 100ms.
```

File: apps/pie-docs/src/components/checkbox/motion/motion.md (hunk 3 above)
Comment (xander-marjoram) [reply apps/pie-docs/src/components/checkbox/motion/motion.md:16]: Discussed on Slack. "Unselected" seems to be more regularly used in the tech context even if "deselected" has a more correct meaning in general.

**Review by xander-marjoram — APPROVED** [review]

**Pull request merged**